pkg/core/native/native_neo.go

```diff
@@ -670,29 +651,32 @@ func makeVoterKey(pub []byte, prealloc ...[]byte) []byte {
 
 // CalculateBonus calculates amount of gas generated for holding value NEO from start to end block
 // and having voted for active committee member.
-func (n *NEO) CalculateBonus(d *dao.Simple, acc util.Uint160, end uint32) (*big.Int, error) {
+func (n *NEO) CalculateBonus(ic *interop.Context, acc util.Uint160, end uint32) (*big.Int, error) {
+	if end != ic.Block.Index {
```

Comment:
We don't always fill the IC's `block` field, thus, I'm afraid this won't work with `CalculatClaimable` RPC call. You have just
```
func (bc *Blockchain) CalculateClaimable(acc util.Uint160, endHeight uint32) (*big.Int, error) {
        ic := bc.newInteropContext(trigger.Application, bc.dao, nil, nil)
        return bc.contracts.NEO.CalculateBonus(ic, acc, endHeight)
}
```
Could you, please, write a test for that?

Reply:
Related to https://github.com/nspcc-dev/neo-go/pull/2892#discussion_r1090860301, @roman-khimov.

Reply:
We actually do have a test for that (`TestWalletClaimGas`) and it's failing.

Reply:
Will do

Reply:
Use `ic.Chain.BlockHeight()+1`？

Reply:
```
if ic.Block != nil && end != ic.Block.Index {
    // boom
}
```
I guess something like this. `else` case is the caller's responsibility and we have a responsible caller in `CalculateClaimable`. Always checking `Chain.BlockHeight()` costs a bit more and `ic.Block.Index` is a bit more logically correct value (even though it'd be the same).

Reply:
Done